src/Microsoft.AspNet.OData.Shared/Query/Validators/OrderByModelLimitationsValidator.cs

```diff
@@ -131,6 +131,24 @@ public override SingleValueNode Visit(NonResourceRangeVariableReferenceNode node
             return null;
         }
 
+        public override SingleValueNode Visit(SingleResourceCastNode nodeIn)
+        {
+            if (nodeIn.Source != null)
+            {
+                if (nodeIn.Source.Kind == QueryNodeKind.SingleComplexNode)
+                {
+                    SingleComplexNode singleComplexNode = nodeIn.Source as SingleComplexNode;
+
+                    if (singleComplexNode != null)
+                    {
+                        return Visit(singleComplexNode);
+                    }
```

Comment:
Having done the check `if (nodeIn.Source.Kind == QueryNodeKind.SingleComplexNode)`, I believe it should be safe to simplify this to:
```suggestion
                    return Visit((SingleComplexNode)node.Source);
```